torch/fx/experimental/sym_node.py

```diff
@@ -240,12 +240,14 @@ def __str__(self):
         return self.str()
 
     def __repr__(self):
-        return (
-            f"SymNode(expr={self.expr}, shape_env={self.shape_env}, "
-            f"pytype={self.pytype}, hint={self._hint}, constant={self.constant})"
-        )
-
-    def _graph_repr(self) -> str:
+        rep = [
+            f"SymNode({self.expr}, shape_env={self.shape_env}, pytype={self.pytype}",
+            *(f"hint={self._hint}" for _ in [None] if self._hint is not None),
+            *(f"constant={self.constant}" for _ in [None] if self.constant is not None),
+        ]
+        return ", ".join(rep) + ")"
+
+    def _graph_repr(self):
```

Comment:
Since this is a new function let's type it properly:
```
def _graph_repr(self) -> str:
```

Reply:
That makes sense. I had this initially, but I realized then that it was raising a linting error due to the method `str` with the same name. To address this issue, I took the liberty of renaming that method to `_str`